biocrnpyler/components_membrane.py

```diff
@@ -0,0 +1,198 @@
+#  See LICENSE file in the project root directory for details.
+
+from typing import List, Union
+
+from .component import Component
+from .reaction import Reaction
+from .species import Complex, Species
+
+class IntegralMembraneProtein(Component):
+    """A class to represent transmembrane proteins or integral membrane proteins.
+    This membrane class is to classify a membrane channel that will intergrate into the membrane.
+    Uses a mechanism called "catalysis".
+    Size is used to indicate number of repeating components to create oligomer. Dimer =2, Trimers = 3, etc."
+    """
+    def __init__(self, membrane_protein: Union[Species, str, Component],
+                 product: Union[Species,str, Component],
+                 direction= None, size=None, attributes=None, **keywords):       
+        """Initialize a MembraneChannel object to store membrane channel related information.
+        :param product: name of the membrane channel, reference to an Species or Component
+        :param direction: transport direction, taken as "Passive"-undirectional unless specified 
+        :param size: number of monomers needed for channel
+        :param attributes: Species attribute
+        :param keywords: pass into the parent's (Component) initializer
+        """
+        
+        # PROTEIN
+        self.membrane_protein = self.set_species(membrane_protein, material_type='protein', attributes=attributes)
+    
+        # PRODUCT is an integrated membrane protein (transmembrane_protein)
+        if product is None:
+            if direction is None:
+                self.product = self.set_species(product, material_type= 'Passive')
+            else:
+                self.product = self.set_species(product, material_type= direction)
+
+        else:
+            if direction is None:
+                self.product = self.set_species(product, material_type= 'Passive')
+            else:
+                self.product = self.set_species(product, material_type= direction)
+        
+        if size is None:
+            self.membrane_protein.size = 1
+        else:
+            self.membrane_protein.size = size
+        
+        Component.__init__(self=self, name=self.membrane_protein.name, **keywords)
+        
+    def get_species(self):
+        return self.membrane_protein
+    
+    def update_species(self):
+        mech_cat = self.get_mechanism('catalysis')
+        
+        return mech_cat.update_species(self.membrane_protein, self.product)
+
+    def update_reactions(self):
+        mech_cat = self.get_mechanism('catalysis')
+        print(self)
+        
+        return mech_cat.update_reactions(self.membrane_protein, self.product, component=self,  part_id=self.name)
+
+class MembraneChannel(Component):
+    """A class to represent Membrane Channels.
+    Assumes the membrane channel transport substrates in a specific direction across the membrane
+    Uses a mechanism called "catalysis"
```

Comment:
I think it would be better for the mechanism type to be called "transport" - otherwise, if we cannot use your transport Mechanisms at the same time a different catalysis mechanism is used (say for metabolism).